### DISCUSSION TITLE
Verification about public blocklists

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#1814] [Q&A]
Hi,
One thing I was entirely sure about in DNScrypt's wiki is way to use public blocklists. Am I right in thinking that the various URLs must be directly added to blocked-ips.txt? Would the below file be correct?
Thanks!
˜Paul

[blocked-ips.txt](https://github.com/DNSCrypt/dnscrypt-proxy/files/6974293/blocked-ips.txt)

**Comment by jedisct1** [comment]
The source code includes this tool: https://github.com/DNSCrypt/dnscrypt-proxy/tree/master/utils/generate-domains-blocklist

Which can load multiple remote and local lists, optimize and merge them to build a single beautiful blocklist.

**Reply by ghost** [reply]
Thanks! I made my way around it and played with the files.
However, when running the python script, I get a large number of similar errors, such as:
[https://gitlab.com/ZeroDot1/CoinBlockerLists/raw/master/list_browser.txt] could not be loaded: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)>
Anything I am doing wrong here?
˜Paul

**Reply by ghost** [reply]
Hi @jedisct1 , any ideas on that certificate failure?
